smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/HttpProtocolTestGenerator.java

```diff
@@ -554,6 +527,76 @@ private void generateResponseTest(OperationShape operation, HttpResponseTestCase
         });
     }
 
+    private void generateServerErrorResponseTest(
+            OperationShape operation,
+            StructureShape error,
+            HttpResponseTestCase testCase
+    ) {
+        Symbol serviceSymbol = serverSymbolProvider.toSymbol(service);
```

Comment:
minor: would be nice to have some more spacing and comments in here describing what's being done.